File: DSRFQ.Web/Modules/Company/Groups/GroupsForm.cs
using Serenity.ComponentModel;
using System;

namespace DSRFQ.Company.Forms;

[FormScript("Company.Groups")]
[BasedOnRow(typeof(GroupsRow), CheckNames = true)]
public class GroupsForm
{
    public string Name { get; set; }
    public string Description { get; set; }
    public ListField<Int32> CompanyList { get; set; }
}

File: DSRFQ.Web/Modules/ServerTypes/Company/GroupsForm.ts
import { StringEditor, LookupEditor, PrefixedContext, initFormType } from "@serenity-is/corelib";

export interface GroupsForm {
    Name: StringEditor;
    Description: StringEditor;
    CompanyList: LookupEditor;
}

export class GroupsForm extends PrefixedContext {
    static readonly formKey = 'Company.Groups';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!GroupsForm.init)  {
            GroupsForm.init = true;

            var w0 = StringEditor;
            var w1 = LookupEditor;

            initFormType(GroupsForm, [
                'Name', w0,
                'Description', w0,
                'CompanyList', w1
            ]);
        }
    }
}